advent_calendar/advent_calendar.html

```diff
@@ -0,0 +1,280 @@
+        { day: 12, type: "video", source: "videos/video_1.mp4" },
+        { day: 13, type: "text", text: "🎉 <strong>Holiday Fun Fact:</strong><br>The first artificial Christmas tree was made in Germany in the 19th century. It was made out of dyed goose feathers!" },
+        { day: 14, type: "image", source: "images/image_7.jpg" },
+        { day: 15, type: "text", text: "🎄 <strong>Christmas Joke:</strong><br>Why did Frosty the Snowman want a divorce? He thought his wife was a little too flaky!" },
+        { day: 16, type: "image", source: "images/image_8.jpg" },
+        { day: 17, type: "video", source: "videos/video_1.mp4" },
+        { day: 18, type: "text", text: "🎅 <strong>Holiday Quote:</strong><br>“Christmas is not a time nor a season, but a state of mind.” – Calvin Coolidge" },
+        { day: 19, type: "image", source: "images/image_9.jpg" },
+        { day: 20, type: "text", text: "🍪 <strong>Vegan Christmas Cookies:</strong><br>Ingredients: 125 g of flour, 120 g of coconut oil (melted), 120 g of maple syrup, 1/4 tsp vanilla extract, 1/4 tsp baking soda, 1/4 tsp salt, optional vegan chocolate chips.<br>Directions: Preheat oven to 180°C. Mix ingredients and roll into balls. Place on a baking sheet and bake for 10-12 minutes. Enjoy!" },
+        { day: 21, type: "image", source: "images/image_10.jpg" },
+        { day: 22, type: "text", text: "🎁 <strong>Vegan Hot Chocolate Recipe:</strong><br>Ingredients: 480 ml of almond milk (or any plant milk), 2 tbsp cocoa powder, 1 tbsp maple syrup, 1/4 tsp vanilla extract, a pinch of salt.<br>Directions: Heat the milk over medium heat. Whisk in cocoa powder, maple syrup, vanilla, and salt until smooth and heated through. Top with vegan whipped cream and marshmallows for extra fun!" },
+        { day: 23, type: "video", source: "videos/video_1.mp4" },
+        { day: 24, type: "video", source: "videos/video_1.mp4" },
+    ];
+
+    // Function to get today's date and restrict access to future doors
+    document.addEventListener("DOMContentLoaded", () => {
+        const today = new Date().getDate();
+        const doors = document.querySelectorAll(".door");
+
+        doors.forEach(door => {
+            const day = parseInt(door.getAttribute("data-day"));
+            //if (day > today) {
+                //door.setAttribute("data-locked", "true");
+                //door.onclick = null;
+            //}
+        });
+    });
+
+    function openContent(day) {
+        const content = calendarContent.find(item => item.day === day);
+        const contentDisplay = document.getElementById("contentDisplay");
+        const contentModal = document.getElementById("contentModal");
+
+        contentDisplay.innerHTML = ""; // Clear previous content
+
+        if (content.type === "video") {
+            contentDisplay.innerHTML = `<video id="modalVideo" controls><source src="${content.source}" type="video/mp4">Your browser does not support the video tag.</video>`;
```

Comment:
I'm with you on this one, if and only if it is a self-made video _AND_ you for some reason can't host it on any site that allows for embedding.

Otherwise see https://www.w3schools.com/tags/tag_iframe.asp.

We do not want to drag around video files.